src/_guides/language/sound-problems.md

```diff
@@ -369,56 +369,99 @@ the command-line Dart VM with `--strong` (where supported).
 
 ### Invalid casts
 
-In Dart 1.x `dynamic` was both a [top type][top_type_wiki] (also known as a
-_universal supertype_, or the _supertype of all other types_) and a
-[bottom type][bottom_type_wiki] (also known as the _zero_ or _empty_ type)
-depending on the context. This meant it was legal to assign, for example,
-an object of type `List<dynamic>` to a `List<String>`:
+Unlike Dart 1.x, Dart 2 enforces a sound type system. This means (roughly)
+you can't get into a situation where the value stored in some variable is
+different than the variable's static type. Like most modern statically
+typed langauges, Dart accomplishes this with a mixture of static
+(compile-time) and dynamic (runtime) checking.
 
-[top_type_wiki]: https://en.wikipedia.org/wiki/Top_type
-[bottom_type_wiki]: https://en.wikipedia.org/wiki/Bottom_type
+For example, this type error is detected at compile-time:
 
-```dart
+{:.fails-sa}
+{% prettify dart %}
 main() {
-  Map json = {'names': []};
-  List<String> names = json['names'];
+  List<int> numbers = [1, 2, 3];
+  List<String> [!string = numbers!]; // <-- Static analysis error
 }
-```
+{% endprettify %}
+ 
+Since neither `List<int>` nor `List<String>` is a subtype of the other,
+Dart rules this out statically. You can see other examples of these
+static analysis errors in [Assigning Mismatched Types](##assigning-mismatched-types).
+
+In other cases, however, Dart needs to insert a _runtime_ check. Consider:
+
+{:.passes-sa}
+{% prettify dart %}
+assumeStrings(List<Object> objects) {
+  List<String> [!strings = objects!]; // <!-- Implicit downcast
```

Comment:
At least drop the `!` from `<!--`.

(The general style used elsewhere in the docs, is IMHO that we avoid using an ASCII arrow prefix like `<--`. If you feel like removing them, great. If not I can sweep the code excerpts later.)